src/Resources/skeleton/registration/RegistrationController.tpl.php

```diff
@@ -7,11 +7,11 @@
 class <?= $class_name; ?> extends <?= $parent_class_name; ?><?= "\n" ?>
 {
 <?php if ($will_verify_email): ?>
-    private $emailVerifier;
+<?= $email_verifier_details->getPropertyStatement() ?>
 
-    public function __construct(EmailVerifier $emailVerifier)
+    public function __construct(<?= $email_verifier_details->getMethodArgument() ?>)
     {
-        $this->emailVerifier = $emailVerifier;
+        <?= $email_verifier_details->getConstructorArgument(true, false) ?>
```

Comment:
I don't see the advantage in these methods of having the property name itself encoded in them. The method is useful for creating the property (which may or may not need a type), and it's even useful (though less-so) for the type-hint on an argument, but I don't see the value in this line (or other lines further down that previously said `$this->emailVerifier...`.

It makes me wonder if a generic helper function just for generating the property type would be simpler:

```php
    private <?= $generator->property($email_verifier_class, 'emailVerifier') ?>;
```

This is faux code - but the first arg would be the FCQN and the 2nd arg would be the property name. 

That still leaves:

> how to type-hint methods in a simpler way without needing to pass 2 variables into the template, one for the fcqn and one for the short name.

We could solve that by passing in the "class name details" objects into the template and using those directly. So, perhaps we pass a "class name details" to the above method. And later for an argument type-hint, we use:

```php
public function __construct(<?= $email_verifier_class->getShortName() ?> $emailVerifier)
```

This would only need really be necessary if a class is dynamic. In most cases, we would just hardcode the class - e.g. `RouterInterface $router`.